lessons/module-3/react-class-components-prework.md

```diff
@@ -0,0 +1,295 @@
+
+- Based on the class definition above, create a new instance of a user. 
+- Using what you already know about objects and methods, utilize the object instance you just created and update the user's age.
+- Using what you already know about objects and methods, utilize the object instance you just created and update the user's email address. 
+
+Tip: use console logs to check that everything is working as you expect. 
+
+<section class="answer">
+
+### A solution _might_ look something like: 
+
+```js
+const henrietta = new User('Henrietta', 40, 'h@gmail.com');
+
+henrietta.updateAge();
+henrietta.updateEmail('newemail@gmail.com');
+```
+</section>
+
+## More Practice
+
+Define a class of your own choice. Make sure it has some properties and at least one method. 
+
+To test it out, instantiate the class (make an object instance). 
+
+## Inheritance
+
+Sometimes, you'll have several classes that are very similar. They may have the same properties or the same functionality. We should work to DRY up our classes, just like we do with any other code. Consider the following class definitions: 
+
+```js
+class Animal {
+  constructor(type, species, weight) {
+    type = type,
+    species = species,
+    weight = weight,
+  }
+
+  makeSound(sound) {
+    console.log(sound);
+  }
+}
+```
+
+```js
+class Dog {
+  constructor(weight, breed) {
+    type = 'Mammal',
+    species = 'Canine',
+    weight = weight,
+    breed = breed,
+  }
+
+  bark() {
+    console.log('woof!');
+  }
+}
+```
+
+- What are all the similarities you see between these two classes? 
+- How does a 'dog' relate to the category of 'animal' in real life? 
+
+<section class="answer">
+
+### Possible answers to the above questions: 
+
+Some similarities between the two classes are: 
+
+- similar properties: type, species, weight
+- similar method: both the method makeSound and bark are doing the same thing.
+
+A dog is a type of animal in real life. A dog could be considered a subcategory of the animal category.
+</section>
+
+This is a perfect time to implement inheritance. Instead of repeating ourselves so much, we can make the Dog class _inherit_ the properties it needs from the Animal class. Then, we could even make more classes that inherit from Animal, such as Cat or Llama. 
+
+We can make a class inherit from a parent class by using the `extends` keyword. We can then call the constructor of the parent class, which will give the Dog class access to the Animal's properties and methods by calling the `super` function. 
+
+```js
+class Animal {
+  constructor(type, species, weight) {
+    this.type = type;
+    this.species = species;
+    this.weight = weight;
+  }
+
+  makeSound(sound) {
+    console.log(sound);
+  }
+}
+
+class Dog extends Animal {
+  constructor(type, species, weight, breed) {
+    super(type, species, weight);
+    this.breed = breed;
+  }
+}
+
+const fido = new Dog('Mammal', 'Canine', 10, 'Yorkshire Terrier');
+fido.makeSound('Woof!');
```

Comment:
Wow!  Cass - this is such an awesome breakdown of classes!   Thank you so much.   I do worry about how much content there is and trying to bring it into Mod 3 without adding more bloat to the curriculum than they can handle.   

I want to consider cutting the inheritance part.  Its really well done so I also kind of want to keep it there as a resource for them.   But I think they can still get enough of an intro without that.